### DISCUSSION TITLE
[Idea] limit access to products and product combinations to specific groups of users

### DISCUSSION THREAD

**Discussion by vnt** [PrestaShop/PrestaShop#37734] [Ideas]
### The problem

display product or product combinations only to specific customer groups
Product A:
• Combinations 1, 2, 3 for groups Visitor, Guest, and Customer.
• Combinations 4, 5, 6 for group B2B.

Product B:
• Combinations 2, 3, 4 for groups Visitor, Guest, and Customer.
• Combinations 3, 4, 5, 6 for group B2B.

Product C for group B2B.

### The solution

Add a field Group Access, as in categories

![Screenshot 2024-12-29 alle 16 37 07](https://github.com/user-attachments/assets/bd452aca-5899-4030-981e-16aa5370fdf8)

### Do you plan to work on this subject?

- [x] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
Hello @vnt 

Having products allowed to be visited by certain groups is quite interesting. Still, combinations are a bit too much, and this would've been a significant change for a bit of use inside our ecosystem.

That said, it's a big change that requires changes in many places, in all places where products are retrieved from the database.

**Comment by vnt** [comment]
Hello kpodemski,
This integration would guarantee an important ability to manage products and variants with respect to user groups.
Just like with categories.
Best regards
